src/sage/topology/simplicial_complex.py

```diff
@@ -4802,7 +4802,85 @@ def intersection(self, other):
             F = F + [s for s in self.faces()[k] if s in other.faces()[k]]
         return SimplicialComplex(F)
 
+    @cached_method
+    def bigraded_betti_numbers(self):
+        r"""
+        Returns a dictionary of the bigraded Betti numbers of ''self'', 
+        with keys '(-i, 2j)'.
+
+        See bigraded_betti_number(self, a, b) for more information.
+
+        EXAMPLES::
 
+            sage: X = SimplicialComplex([[0,1],[1,2],[1,3],[2,3]])
+            sage: Y = SimplicialComplex([[1,2,3],[1,2,4],[3,5],[4,5]])
+            sage: X.bigraded_betti_numbers()
+            {(0, 0): 1, (-1,4 ): 2, (-1, 6): 1, (-2, 6): 1, (-2, 8): 1}
+            sage: Y.bigraded_betti_numbers()
+            {(0, 0): 1, (-1,4 ): 3, (-2, 6): 1, (-2, 8): 2, (-3, 10): 1}
+        """
+        from sage.misc.functional import gens 
+        from sage.homology.homology_group import HomologyGroup
+        L = set(self.vertices())
+        n = len(L)
+        B = {}
+        H0 = HomologyGroup(0, ZZ)
+
+        for i in range(n+1):
+            for j in range (n+1):
+                B[(-i, 2*j)] = 0
```

Comment:
I would not preinitialize the `B` dict to zero values. Check to see if the corresponding key is already there when you manipulate it.

Reply:
I agree, setting them to ```ZZ.zero()``` during the main loop, as you suggested, is a better approach. I just initialized ```B[(0,0)]``` to ```ZZ.one()```. For some reason, this does not evaluate properly in the main loop, but when i call ```bigraded_betti_numbers(0,0)```, it does return ```1```. I can't seem to find out why, but I will keep looking into it.